src/figuresClasses.ts

```diff
@@ -1,19 +1,87 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  color: Color;
+  shape: Shape;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  constructor(
+    public color: Color,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+    public shape: Shape = 'triangle',
+  ) {
+    this.checkIfTriangle();
+  }
 
-}
+  checkIfTriangle(): void {
+    const sidesOfTriangleArr = [this.sideA, this.sideB, this.sideC]
+      .sort((a: number, b: number) => a - b);
+
+    if (Math.min(...sidesOfTriangleArr) <= 0) {
+      throw new Error('All sides have to be positive numbers');
+    }
+
+    if (sidesOfTriangleArr[2]
+      >= (sidesOfTriangleArr[0] + sidesOfTriangleArr[1])) {
+      throw new Error('This is not a triangle');
+    }
+  }
 
-export class Circle {
+  getArea(): number {
+    const p = (this.sideA + this.sideB + this.sideC) / 2;
+    const areaOfTriangle = Math
+      .sqrt(p * (p - this.sideA) * (p - this.sideB) * (p - this.sideC));
 
```

Comment:
area will do, as you are currently inside the class and it is pretty clear what we're dealing with